include/nfl/poly.hpp

```diff
@@ -99,17 +100,17 @@ class poly {
   poly(uniform const& mode);
   poly(non_uniform const& mode);
   template <class in_class, unsigned _lu_depth> poly(gaussian<in_class, T, _lu_depth> const& mode);
-  poly(std::initializer_list<value_type> values, bool reduce_coeffs = true);
-  template <class It> poly(It first, It last, bool reduce_coeffs = true);
+  poly(std::initializer_list<value_type> values, size_t length = 0, bool reduce_coeffs = true);
+  template <class It> poly(It first, It last, size_t length = 0, bool reduce_coeffs = true);
   template<class Op, class... Args> poly(ops::expr<Op, Args...> const& expr);
 
   void set(value_type v, bool reduce_coeffs = true);
   void set(uniform const& mode);
   void set(non_uniform const& mode);
   template <class in_class, unsigned _lu_depth> void set(gaussian<in_class, T, _lu_depth> const& mode);
   void set(void* mode);
-  void set(std::initializer_list<value_type> values, bool reduce_coeffs = true);
-  template <class It> void set(It first, It last, bool reduce_coeffs = true);
+  void set(std::initializer_list<value_type> values, size_t length = 0, bool reduce_coeffs = true);
+  template <class It> void set(It first, It last, size_t length = 0, bool reduce_coeffs = true);
```

Comment:
I find this API confusing. As the behavior changes when length is set or not, I would advocate for two different functions here, one with length and one without. Eventually make an auxillary function to syndicate code afterward